app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb

```diff
@@ -18,8 +17,12 @@ def initialize
         )
       end
 
-      def extract_token_from_credentials(credentials)
-        @extract_token_from_credentials.call(credentials: credentials)
+      def create_authentication_parameters(authenticator_input)
+        @authentication_parameters = Authentication::AuthnJwt::AuthenticationParameters.new(authenticator_input)
+        @authentication_parameters.credentials = authenticator_input.request.read
+        @authentication_parameters.jwt_token = @extract_token_from_credentials.call(
```

Comment:
can this be done in the `AuthenticationParameters` init method?

Reply:
this piece of code indicates that something in the design is not tight

Reply:
>this piece of code indicates that something in the design is not tight
@orenbm Can you please elaborate more

Reply:
you have an `authentication_parameters` object that has a `jwt_token` object that relies on the `authentication_parameters` object that contains it. it is a circular dependency.

I _think_ that this logic can be done inside `AuthenticationParameters` as both `credentials` and `jwt_token` need the `authenticator_input.request.read` which is already passed to the class

Reply:
I dont have a strong opinion here, but if i must to choose, i prefer the POJO approach to leave the `AuthenticationParameters` class without external logic